code/game/objects/items/breakable.dm

```diff
@@ -0,0 +1,260 @@
+	if(breakable_flags && break_item(target,range,speed,override,fly_speed))
+		return
+	else
+		..()
+
+/////////////////////
+
+//Breaking items:
+
+//Attacking the item with a wielded weapon or other item
+/obj/item/attackby(obj/item/W, mob/user)
+	if(isobserver(user) || !Adjacent(user) || user.is_in_modules(src))
+		return
+
+	if(user.a_intent == I_HURT && breakable_flags & BREAKABLE_WEAPON && loc != user) //Smash items on the ground, but not in your inventory.
+		if(!isnull(breakable_exclude)) //Check that the weapon isn't specifically excluded from hitting this item
+			for(var/obj/item/this_excl in breakable_exclude)
+				if(istype(W,this_excl))
+					return ..()
+		user.do_attack_animation(src, W)
+		user.delayNextAttack(1 SECONDS)
+		add_fingerprint(user)
+		var/glanced=!take_damage(W.force)
+		user.visible_message("<span class='warning'>\The [user] [pick(W.attack_verb)] \the [src] with \the [W][generate_break_text(glanced,TRUE)]</span>","<span class='notice'>You hit \the [src] with \the [W][generate_break_text(glanced)]<span>")
+		break_item()
+		//Break the weapon as well, if applicable, based on its own force.
+		if(W.breakable_flags & BREAKABLE_AS_MELEE)
+			W.take_damage(min(W.force, BREAKARMOR_MEDIUM),FALSE) //Cap it at BREAKARMOR_MEDIUM to avoid a powerful weapon also needing really strong armor to avoid breaking apart when used.
+			W.break_item()
+		return
+	else
+		..()
+
+//Simple animals attacking the item
+/obj/item/attack_animal(mob/living/simple_animal/M)
+	if(M.melee_damage_upper && M.a_intent == I_HURT && breakable_flags & BREAKABLE_UNARMED)
+		M.do_attack_animation(src, M)
+		M.delayNextAttack(1 SECONDS)
+		var/glanced=!take_damage(rand(M.melee_damage_lower,M.melee_damage_upper))
+		M.visible_message("<span class='warning'>\The [M] [M.attacktext] \the [src][generate_break_text(glanced,TRUE)]</span>","<span class='notice'>You hit \the [src][generate_break_text(glanced)]</span>")
+		break_item()
+	else
+		..()
+
+//Item ballistically colliding with something
+
+/obj/item/throw_impact(atom/hit_atom, var/speed, mob/user)
+	..()
+	if(!(breakable_flags & BREAKABLE_AS_THROWN))
+		return
+	if(!(breakable_flags & BREAKABLE_MOB) && istype(hit_atom, /mob)) //Don't break when it hits a mob if it's not flagged with BREAKABLE_MOB
+		return
+	if(isturf(loc)) //Don't take damage if it was caught mid-flight.
+		//Unless the item falls to the floor unobstructed, impacts happens twice, once when it hits the target, and once when it falls to the floor.
+		var/thisdmg = 5 * get_total_scaled_w_class(1) / speed //impact damage scales with the weight class and speed of the item. since a smaller speed is faster, it's a divisor.
+		if(istype(hit_atom, /turf/simulated/floor))
+			take_damage(thisdmg/2)
+		else
+			take_damage(thisdmg, FALSE) //Be verbose about the item taking damage.
+	break_item(null,null,null,null,null,hit_atom)
```

Comment:
Probably use a named arg instead

Reply:
> Also, the code style is very inconsistent and bad, but since we _still_ don't have an official style guide, I don't get to officially complain about it.

Just for my own edification, what kind of things would be good practice in terms of code style?